Automattic/wp-calypso
# Make product variants in the lightbox URL accessible.

### Description
Supporting URL accessibility for product variants shown in the Lightbox is the primary objective of this PR. I took the opportunity to enhance the way we now activate the product lightbox component. I renamed `useItemLightbox` to `useProductLightbox` in order to better complement the primary component `ProductLightbox` for which it was originally created.

#### Proposed Changes

* [Remove the currentProduct state within the lightbox component and use an external source](https://github.com/Automattic/wp-calypso/commit/273b72d843ce2ba06409b12315ed80168fd44c82). This will make it easier for us to ensure the getters and setters for currentProduct state are coming from the `useProductLightbox` hook. Having the logic for setting the current product within the `useProductLightbox` allows us to hide the implementation of manipulating location hash from consuming components.

* [Rename useItemLightbox to useProductLightbox for better context ](https://github.com/Automattic/wp-calypso/pull/67791/commits/8fb213beeb8fee5d0856349425b97639c921442d) and compliment the `ProductLightbox` component name.

#### Testing Instructions



* 
1. * Run `git fetch && git checkout add/lightbox-url-accessible-product-variant`
    * Run `yarn start-jetpack-cloud`
3. Go to http://jetpack.cloud.localhost:3000/pricing?flags=jetpack/pricing-page-rework-v1 or use the jetpack cloud live link and goto `/pricing?flags=jetpack/pricing-page-rework-v1`.
4. Click on Backup featured or simple card's more info link.
5. Select the 1TB Options
<img width="358" alt="Screen Shot 2022-09-14 at 6 02 10 PM" src="https://user-images.githubusercontent.com/56598660/190125123-18d5de49-3ff4-448a-b2d1-921c6c2469ec.png">

6. Confirm that `jetpack_backup_t2_yearly` is set in the URL.
<img width="222" alt="Screen Shot 2022-09-14 at 6 03 35 PM" src="https://user-images.githubusercontent.com/56598660/190125237-46d23180-0850-45a9-8521-aa2eb2a494d6.png">

7. Copy/paste the link in a new browser tab.
8. Confirm that the lightbox loads the correct variant during on load.

<img width="1115" alt="Screen Shot 2022-09-14 at 6 05 17 PM" src="https://user-images.githubusercontent.com/56598660/190125786-965a7a4f-7613-485e-ad7a-ee4ec9d4ea39.png">



#### Pre-merge Checklist






Related to 1202796695664022-as-1202936358291308
